language-plutus-core/plutus-ir/Language/PlutusIR/Normalize.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE FlexibleContexts #-}
+-- | The user-facing API of the normalizer.
+
+module Language.PlutusIR.Normalize
+    ( Export.normalizeType
+    , normalizeTypesIn
+    , normalizeTypesInProgram
+    ) where
+
+import  Language.PlutusCore.Core as PLC (Normalized (..))
+import           Language.PlutusCore.Name
+import           Language.PlutusCore.Normalize.Internal hiding (normalizeTypesInM)
+import           Language.PlutusCore.Quote
+import Language.PlutusIR.Transform.Rename () -- for instances
+import Language.PlutusCore.Rename (rename)
+import Language.PlutusIR
+import Language.PlutusCore.Normalize as Export (normalizeType)
+
+import           Control.Lens
+import           Control.Monad                          ((>=>))
+
+-- | Normalize every 'Type' in a 'Term'.
+normalizeTypesIn
+    :: (HasUnique tyname TypeUnique, HasUnique name TermUnique, MonadQuote m)
+    => Term tyname name uni ann -> m (Term tyname name uni ann)
+normalizeTypesIn = rename >=> runNormalizeTypeM . normalizeTypesInM
```

Comment:
The whole Normalize and Normalize/Internal were ported from plutus-core to plutus-ir. The PlutusIR refers to PIR.Term , where plutuscore uses PLC.Term.  There must be a better way to avoid this duplication.

Reply:
I think only the interface that deals with terms is duplicated, and that does need to be different, since as Nikos says, the terms are different.

Reply:
Yeah, sorry, I can't read and for some reason thought this function was normalizing a `Type`. I agree that duplication here is perfectly fine.